### DISCUSSION TITLE
Make addresses optional until they become necessary

### DISCUSSION THREAD

**Discussion by andreaskienast** [shopware/platform#3162] [RFC / Proposal]
**Effort**: high

### Description: 
We're using Shopware as a 3rd party system connected to a middleware that allows a user to have no addresses at all. It's currently a heavy burden for us to enforce at least a stub address at the correct place as Shopware requires such under any circumstances, otherwise, we're facing exceptions everywhere. Hence my proposal is to make addresses optional until any occurrence makes it necessary, e.g. the checkout. This should be intercepted somehow (perhaps by a globally catched exception?) that renders a UI to let the user add an address. After that, the user should be able to continue with the original request.

### Benefits
This makes integration of Shopware was a 3rd party application into an existing system much easier.

### Break strategy
Hide this feature behind a feature flag to give plugin developers the opportunity to adapt to this behavior.

**Comment by phinpope** [comment]
We have all our users with an address in an imaginary country so that we can create accounts via API and link them to our middleware sign on accounts.
